src/main/java/uk/ac/ebi/biostudies/service/impl/BatchDownloadScriptBuilder.java

```diff
@@ -18,53 +19,55 @@
+                String fileTemplate = IOUtils.toString(templateStream, StandardCharsets.UTF_8);
+                DownloadTemplates.put(fName, fileTemplate);
+                templateStream.close();
+            } catch (Exception ex) {
+                LOGGER.error("Cant open download template file {}", fName, ex);
+            }
+        }
+    }
+
+    public String fillTemplate(String downloadType, List<String> fileNames, String baseDirectory, String os, Constants.File.StorageMode storageMode) {
+        String content = "";
+        try {
             String fileTemplate = DownloadTemplates.get(getTemplate(downloadType, os));
             content = fillFileTemplate(fileTemplate, fileNames, baseDirectory, downloadType, storageMode);
-            if(!os.equalsIgnoreCase(Constants.OS.WINDOWS))
-                content = "#!/bin/bash\r\n"+content;
-        }catch (Exception ex){
+            if (!os.equalsIgnoreCase(Constants.OS.WINDOWS))
+                content = "#!/bin/bash\r\n" + content;
+        } catch (Exception ex) {
             LOGGER.error("Cant open download template file {}", getTemplate(downloadType, os), ex);
         }
         return content;
     }
-    String getTemplate(String type, String os){
+
+    String getTemplate(String type, String os) {
         String myOs = os;
-        if(!myOs.equalsIgnoreCase(Constants.OS.WINDOWS))
-            myOs=Constants.OS.UNIX;
+        if (!myOs.equalsIgnoreCase(Constants.OS.WINDOWS))
+            myOs = Constants.OS.UNIX;
         StringBuilder result = new StringBuilder(type);
         result.append("-").append(myOs);
         return result.toString();
     }
 
-    String fillFileTemplate(String fileTemplate, List<String> fileNames, String baseDirectory, String downloadType, Constants.File.StorageMode storageMode){
+    String fillFileTemplate(String fileTemplate, List<String> fileNames, String baseDirectory, String downloadType, Constants.File.StorageMode storageMode) {
         String content = "";
-        if(downloadType.equalsIgnoreCase("ftp")){
-            String allFiles = fileNames.stream().map(name -> "mget \""+name+"\"").collect(Collectors.joining("\r\n"));
+        if (downloadType.equalsIgnoreCase("ftp")) {
+            String allFiles = fileNames.stream().map(name -> "mget \"" + name + "\"").collect(Collectors.joining("\r\n"));
             content = String.format(fileTemplate, storageMode.toString().toLowerCase() + "/" + baseDirectory, allFiles);
-        }
-        else if(downloadType.equalsIgnoreCase("aspera")){
-            content = fileNames.stream().map(name -> String.format(fileTemplate, "\""+baseDirectory+"/Files/"+name+"\"")).collect(Collectors.joining("\r\n"));
+        } else if (downloadType.equalsIgnoreCase("aspera")) {
+            content = fileNames.stream().map(name -> String.format(fileTemplate, "\"" + storageMode.toString().toLowerCase() + "/" + baseDirectory + "/Files/" + name + "\"")).collect(Collectors.joining("\r\n"));
```

Comment:
What is a proper aspera directory? is it like this: fire/baseDirectory/files ...